File: 002.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import sys

def fib_slow(n):
    if n == 1:
        return 1
    elif n == 2:
        return 2
    else:
        return fib_slow(n - 1) + fib_slow(n - 2)

def fib_gen(n):
    i = 1
    while True:
        x = fib_slow(i)
        if x > n:
            return
        yield x
        i += 1

def p2(n):
    return sum(filter(lambda x: x % 2 == 0, [ x for x in fib_gen(n) ]))

def main():
    print(p2(4000000))

if __name__ == '__main__':
    main()

File: elixir/002.exs
#!/usr/bin/env elixir

# https://projecteuler.net/problem=2

defmodule P2 do
	def fib(max, [h|t]) when h > max do
		t
	end
	def fib(max, l) do
		[a|t] = l
		[b|_] = t
		fib(max, [a+b|l])
	end
	def fib(max) do
		fib(max, [2, 1])
	end
end

P2.fib(4000000)
|> Enum.filter(&(rem(&1, 2) == 0))
|> Enum.sum
|> IO.inspect
